media/js/mozorg/home/home.js

```diff
@@ -2,147 +2,203 @@
+
+     }
+
+     function getNextEl(el) {
+         el = el.nextSibling;
+         while (el) {
+             if (el.nodeType === 1) {
+                 return el;
+             }
+             el = el.nextSibling;
+         }
+         return null;
+     }
 
     /*
      * Video Tracking
      */
 
-    function trackVideoInteraction(title, state) {
-        window.dataLayer.push({
-            'event': 'video-interaction',
-            'videoTitle': title,
-            'interaction': state
-        });
-    }
+     function trackVideoInteraction(title, state) {
+         window.dataLayer.push({
+             'event': 'video-interaction',
+             'videoTitle': title,
+             'interaction': state
+         });
+     }
 
-    function initVideoEvents() {
-        var videos = document.querySelectorAll('video');
-        var videoCards = document.querySelectorAll('.mzp-c-card.has-video-embed .mzp-c-card-block-link');
+     function initVideoEvents() {
+         var videos = document.querySelectorAll('video');
+         var videoCards = document.querySelectorAll('.mzp-c-card.has-video-embed .mzp-c-card-block-link');
 
         // open and play videos in a modal on click.
-        for (var i = 0; i < videoCards.length; i++) {
-            videoCards[i].addEventListener('click', playVideo, false);
-        }
+         for (var i = 0; i < videoCards.length; i++) {
+             videoCards[i].addEventListener('click', playVideo, false);
+         }
 
         // track video interaction events for GA.
-        for (var j = 0; j < videos.length; j++) {
-            videos[j].addEventListener('play', function() {
-                trackVideoInteraction(this.getAttribute('data-ga-label'), 'play');
-            }, false);
-
-            videos[j].addEventListener('pause', function() {
-                var action = this.currentTime === this.duration ? 'complete' : 'pause';
-                trackVideoInteraction(this.getAttribute('data-ga-label'), action);
-            }, false);
-        }
-    }
-
-    function playVideo(e) {
-        var card = e.currentTarget;
-        var title = card.querySelector('.mzp-c-card-title').innerText;
-        var sibling = getNextEl(e.currentTarget);
-        var content = sibling.querySelector('.mzp-c-card-video-content');
-
-        if (content) {
-            e.preventDefault();
-            var video = content.querySelector('video');
-
-            Mozilla.Modal.createModal(this, content, {
-                title: title,
-                className: 'mzp-has-media',
-                onCreate: function() {
-                    try {
-                        video.load();
-                        video.play();
-                    } catch(err) {
-                        // fail silently
-                    }
-                },
-                onDestroy: function() {
-                    video.pause();
-                }
-            });
-        }
-    }
+         for (var j = 0; j < videos.length; j++) {
```

Comment:
Since we want to track play/pause events using the YouTube API, we can probably remove the old event listeners here.